### DISCUSSION TITLE
How to exchange session token(session api) for a JWT?

### DISCUSSION THREAD

**Discussion by vkhobor** [zitadel/zitadel#8871] [Q&A]
In a custom login I would like to authenticate with the session api, but on my resource server I would like to authenticate with only checking a JWT, or without calling out Zitadel on every request. 

Is this possible?
Can I exchange the Session token with to a JWT somehow?

**Comment by hifabienne** [comment]
Hi @vkhobor 
Currently this is not possible.
We do have an open issue for that, if you would like to see this implemented, please upvote it.
https://github.com/zitadel/zitadel/issues/7900

**Reply by vkhobor** [reply]
Thanks, upvoted.

So for a custom UI until this lands the recommended is to fetch the session on every request, is it?

**Reply by robinjhector** [reply]
This seems strange. JWTs is such an established paradigm for how to represent a signed in user, it's weird that Zitadel doesn't support this out of the box

**Reply by fforootd** [reply]
Need to check with the team what the reason was to default to opaque tokens in this case... maybe @livio-a knows this from the top of his head.

On other points of Zitadel we support using both types though. Like you can enable JWT when using Zitadel with OIDC/Oauth.
